helpers/embeds.js

```diff
@@ -0,0 +1,78 @@
+import { findComposedAncestor } from '../helpers/dom.js';
+import { html } from 'lit';
+import { tryGetPluginByKey } from '@brightspace-ui/core/helpers/plugins.js';
+
+class RenderNode {
```

Comment:
This is just a little helper to encapsulate the rendering logic and make sure we're tracking/following child relationships properly. It's probably not needed, and there may be cleaner ways to handle this, but for now I at least like that it makes debugging easier.